### PR TITLE
Add an nginx https example

### PR DESCRIPTION
**Pull request by bprashanth**
I'm going to use this container/key generation makefile/secret generation helper script in the user guides.
@bgrant0607

### CODE REVIEW

**Comment by bprashanth** [comment]
Used by https://github.com/GoogleCloudPlatform/kubernetes/pull/11267
@thockin

File: examples/nginx-https/README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<h1>*** PLEASE NOTE: This document applies to the HEAD of the source
```
Comment (thockin) [inline examples/nginx-https/README.md]: you'll need to re-run gendocs - this header changed

File: examples/nginx-https/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+First generate a self signed rsa key and certificate that the server can use for TLS.
+
+```shell
+$ make keys secret KEY=/tmp/nginx.key CERT=/tmp/nginx.crt SECRET=/tmp/secret.json
+```
+
+### Run the nginx https server in a docker container
+
+This step is optional. The [docker daemon](https://docs.docker.com/userguide/) needs to be running on your host for this to work.
+Start the nginxhttps server and mount the generated keys into the container.
```
Comment (thockin) [inline examples/nginx-https/README.md]: "nginx https" ?

File: examples/nginx-https/README.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+tree only. If you are using a released version of Kubernetes, you almost
+certainly want the docs that go with that version.</h1>
+
+<strong>Documentation for specific releases can be found at
+[releases.k8s.io](http://releases.k8s.io).</strong>
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+# Nginx https service
```
Comment (thockin) [inline examples/nginx-https/README.md]: how is this different from simple-nginx ?  Is it worth having 2 examples?

File: examples/nginx-https/README.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+```shell
+$ make keys secret KEY=/tmp/nginx.key CERT=/tmp/nginx.crt SECRET=/tmp/secret.json
+```
+
+### Run the nginx https server in a docker container
+
+This step is optional. The [docker daemon](https://docs.docker.com/userguide/) needs to be running on your host for this to work.
+Start the nginxhttps server and mount the generated keys into the container.
+```
+$ docker run -P -d --name mynginx -v "/tmp/nginx.crt:/etc/nginx/ssl/nginx.crt" -v "/tmp/nginx.key:/etc/nginx/ssl/nginx.key" nginxhttps
```
Comment (thockin) [inline examples/nginx-https/README.md]: It's a little squicky that we have raw docker commands going on.  Can we do without?  I know certs is weird.

File: examples/nginx-https/README.md (hunk 3 above)
Comment (bprashanth) [inline examples/nginx-https/README.md]: we could merge the 2, simplenginx uses the standard nginx container which listens on port 80 and doesn't understand https. The real point of this example is to include it in a user guide where we describe multi-port (http/https) services. This README is more about having a working https server and less about running nginx in a container.

Comment (bprashanth) [inline examples/nginx-https/README.md]: If we decide not to merge, I should probably rename to https-nginx

File: examples/nginx-https/README.md (hunk 4 above)
Comment (bprashanth) [inline examples/nginx-https/README.md]: I was trying to bridge the gap between secrets/volumes and how one would do that with docker. Can surely remove.

**Comment by bprashanth** [comment]
@thockin removed unnecessary stanza and regenerated docs, waiting on shippable

**Comment by bprashanth** [comment]
ick
```
Successful get pods {{range.items}}{{.metadata.name}}:{{end}}:
the path "examples/multi-pod.yaml" does not exist
!!! Error in ./hack/test-cmd.sh:421
```
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/user-guide/multi-pod.yaml

**Comment by bgrant0607** [comment]
all green

**Comment by thockin** [comment]
LGTM

medium term we should have 1 nginx example in multiple parts.

**Pull request merged**